packages/logic-utils/src/utils/flatten-members.ts

```diff
@@ -10,7 +10,13 @@ function flattenMembers<T>(
   if (ui5SuperClass !== undefined) {
     // UI5 SDK refers to inherited members (properties, events, aggregations. ...) as "borrowed" ...
     const borrowedMembers = flattenMembers(membersGetter, ui5SuperClass);
-    return directMembers.concat(borrowedMembers);
+    return directMembers.concat(
```

Comment:
Could you please refactor this logic with two separate steps
1. **one** step to created "borrowedAndNotOverridden" (better name?) list and assign it to a **separate** variable..
2. **second** step to add this list to the result

Also in this project we tend to use `lodash` utilities and not built-ins (.find / .filter) even
if those built-ins exist and are safe to use.

This is to ensure consistent style of writing logic ([].find(() => {}) vs _.find([], () => {})
You can see example in the rest of the code (e.g in logic-utils)